### ISSUE TITLE
No typing echo from VoiceOver for Mac when writing

### ISSUE THREAD

**Issue by extra808** [ckeditor/ckeditor5#15436]
## 📝 Provide detailed reproduction steps (if any)

1. Turn on VoiceOver, press VO-V and set Typing Echo to "Characters and Words" (the default)
2. In Safari for Mac, open a page with CKEditor
3. Move focus to the editing area and type content

### ✔️ Expected result

VoiceOver repeats each letter as it's typed, words are spoken once they're finished.

### ❌ Actual result

VoiceOver says nothing while typing, not letters and not words.

### ❓ Possible solution

Not a solution but a crude workaround: VoiceOver will read text that's already in the editor, one can type then select all the text in the editor to hear it.

## 📃 Other details

* Browser: Safari 17.1
* OS: macOS 13.6.2 (Ventura)
* First affected CKEditor version: Unknown, have reproduced the problem locally in 38.1.0, in [CKEditor5 Predefined builds](https://ckeditor.com/docs/ckeditor5/latest/examples/builds/multi-root-editor.html), but not in [CKEditor 4 demo](https://ckeditor.com/ckeditor-4/demo/).

I can also reproduce the problem using Safari Technology Preview Release 183 and with Safari 16.5 on macOS 12.6 but haven't reproduced it with any browser that isn't Safari for Mac. I cannot reproduce the problem using VoiceOver with Safari on iOS 17.1 (with virtual or Bluetooth keyboard), VoiceOver with Firefox or Chrome for Mac, or NVDA with Firefox on Windows 10.

**Comment by extra808** [comment]
The problem persists in Safari 17.1.2 on macOS 14.1.2 (Sonoma).

**Comment by oleq** [comment]
Does this reproduce in a raw `contenteditable` element like this one here https://codepen.io/anow_cks/pen/dyaLEVV ? Because if that's the case, I'm afraid there's nothing we can do about it and it's entirely on Apple's side. 

On a side note, VoiceOver has been known for its erratic behavior and missing functionality. In my experience, NVDA and JAWS are way more advanced in terms of document structure and ARIA attribute recognition.

**Comment by rpkoller** [comment]
I've tested with Safari 17.2 and Sonoma 14.2 in Drupal 10.1.7 (was able to reproduce the problem with previous versions of sonoma and safari as well). The problem presists, there is no typing echo. I've then tested the pen @oleq provided, there i get a typing echo.  

https://github.com/ckeditor/ckeditor5/assets/1665422/8d2189eb-ab4e-44eb-ab73-550ca364d679

**Comment by niegowski** [comment]
This might be related to the way we handle the `beforeinput`. We prevent the browser behavior and manually handle the model modification. I suspect that this might affect this behavior same as auto text replacement in Safari.

**Comment by oleq** [comment]
> This might be related to the way we handle the `beforeinput`. We prevent the browser behavior and manually handle the model modification. I suspect that this might affect this behavior same as auto text replacement in Safari.

This is a good point. How do we go about checking this?

**Comment by Witoso** [comment]
Rel: #13428.
